### ISSUE TITLE
Make every scope a class.

### ISSUE THREAD

**Issue by pradyunsg** [FichteFoll/CSScheme#13]
Much like Atom syntaxes, it would be nice if the first scope would (optionally, because backwards compatibility) be a class-name instead of a tag name. This would require some changes here and there, but I feel it might be worth it because then it would be possible to support both Atom and Sublime Text with the exact same LESS code. Yay!

Anyone thinks this might be a good idea.

Sidenote:
Why is this package not used more!? Please advice/advertise. :wink:

**Comment by FichteFoll** [comment]
Does Atom care about the order of scope parts, which we interpret as "classes" in CSScheme? What I mean is, does Atom differentiate between `.source.css` and `.css.source`? If it does not, than I would prefer not to "dumb it down" to Atoms level but rather add a compatibility layer that omits leading dots for each scope name.

... We really need some good terminology for all parts of a "scope".

> Why is this package not used more!? Please advice/advertise. 

Probably exacly because of that: bad advertisement. I try to mention it whenever it becomes relevant, but I wonder if others do as well.

**Comment by pradyunsg** [comment]
> What I mean is, does Atom differentiate between .source.css and .css.source?

Nope. I don't think we want any smartness in this code. It just converts the CSS to plist files.

**Comment by pradyunsg** [comment]
> ... We really need some good terminology for all parts of a "scope".

I'll give it a shot... I've borrowed from TextMate where I felt it was useful.

---

```
source.python.3 meta.function.python.3 variable.parameter.function.python.3
```
- Selector - The entire thing
- Element - Any portion as separated by spaces. eg. `meta.function.python.3`
- Group - A dot-separated item in an element. eg. `meta` or `function` or `python` or `3`.
  - Root group - The first group of an element. (because they're standardized)

Of course, because we might need to refer to elements in relation to each other, we can use the Parent-Child relationship. 
- In the above selector,
  - `meta.function.python.3` is the child of `source.python.3`
  - `meta.function.python.3` is the parent of `variable.parameter.function.python.3`.
- In `meta.function.python.3`,
  - `function` is the parent group of `python`
  - `function` is the child group of `meta`.

---

I suggest adding whatever would be finalized to this Project's Wiki.

**Comment by pradyunsg** [comment]
Bump.

**Comment by pradyunsg** [comment]
Closing this out since it seems unlikely that either of us are gonna work on this. :)

**Comment by FichteFoll** [comment]
All right, I didn't have a good argument for it in the first place.
